.github/workflows/build.yml

```diff
@@ -21,37 +21,70 @@ jobs:
     runs-on: windows-latest
     steps:
     - uses: actions/checkout@v1
+    - name: Check for source changes
+      run: |
+        if [ -z "GITHUB_BASE_REF" ]; then
+          echo '::set-env name=run_tests::true'
+        else
+          git diff --name-only origin/$GITHUB_BASE_REF... | grep -qvE '(\.rst$|^Doc|^Misc)' && echo '::set-env name=run_tests::true'
+        fi
+      shell: bash
     - name: Build CPython
+      if: env.run_tests
       run: .\PCbuild\build.bat -e -p Win32
     - name: Display build info
+      if: env.run_tests
       run: .\python.bat -m test.pythoninfo
     - name: Tests
+      if: env.run_tests
       run: .\PCbuild\rt.bat -q -uall -u-cpu -rwW --slowest --timeout=1200 -j0
 
   build_win_amd64:
     name: 'Windows (x64)'
     runs-on: windows-latest
     steps:
     - uses: actions/checkout@v1
+    - name: Check for source changes
+      run: |
```

Comment:
Instead of duplicating this check for every job, could we have a single job that checks the documentation-only status and then use [`job.needs`](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds) to pass the information to the other ones?

Reply:
But that would be a completely different job, and does not keep env variables AFAIK.

Reply:
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjobs_idoutputs

Reply:
See https://github.community/t5/GitHub-Actions/Sharing-a-variable-between-jobs/td-p/38880 for context. It appears the only way to share data between jobs is by uploading artifacts, which is a bit overkill in this case.

Github actions does not support reusing steps either right now, which is a shame.

Reply:
You might have already seen my comment above but it looks like this is a relatively new feature: https://github.community/t5/GitHub-Actions/Sharing-a-variable-between-jobs/m-p/54424/highlight/true#

Reply:
Yes, sorry. I was not aware this was possible, I have been in similar situations before and always had to use workarounds :simile: